src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -227,6 +227,22 @@ private function parseDefaults(array &$content, $file)
     }
 
     /**
+     * @param array $service
+     *
+     * @return bool
+     */
+    private function isUsingShortSyntax(array $service)
+    {
+        foreach ($service as $key => $value) {
+            if (!is_int($key) && (isset($key[0]) && '$' !== $key[0])) {
```

Comment:
the empty string case should also return false, isn'it? thus:
`if (is_string($key) && ('' === $key || '$' !== $key[0])) {`